include/libdnf/transaction/transaction.hpp

```diff
@@ -242,11 +241,25 @@ class Transaction {
     /// Fill the transaction packages.
     void fill_transaction_packages(const std::vector<libdnf::base::TransactionPackage> & transaction_packages);
 
+    /// Start the transaction by inserting it into the database
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.begin()
+    void start();
+
+    /// Finish the transaction by updating it's state in the database
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.finish(libdnf::TransactionState state)
+    void finish(TransactionState state);
```

Comment:
Why `start()` and `finish()` is public?

Reply:
They were public already, I just moved the code to have a nicer method order.

Reply:
Yes, but I think they should be private. Only DNF internels should allow to store transaction and anly transaction created internally without user modification + - elements. Only adding comments or transaction label should be allowed and may be with indirect modification.

Reply:
I tend to agree with @j-mracek here. I think `transaction::Transaction` objects should be read-only for API users. Is there a use case where creating / modifying Transaction objects via API is needed? History database records should be only created automatically during execution of `base.Transaction.run()`. It sounds weird  to me that API users can mess with the history database by creating new records there haphazardly.
Do you think this is a problem that deserves a TODO and a tracking ticket?

Reply:
I agree as well. It's on my list of things to do to finalize the API, I've made a note in the ticket we have for that and I've also added a TODO to a semi-random point in the code :slightly_smiling_face: